### PR TITLE
Swig

### PR DESCRIPTION
**Pull request by ocefpaf**
Just testing...

Ping @gillins

### CODE REVIEW

File: recipe/meta.yaml [hunk 1]
```diff
@@ -18,12 +18,14 @@ source:
...
 build:
-  number: 2
+  number: 3
 
 requirements:
   build:
     - python
+    - swig
     - setuptools
+    - posix  # [win]
```
Comment (ocefpaf) [inline recipe/meta.yaml]: Not sure if we need that to run `make`. @mingwandroid is that correct? (Sorry to keep bothering you with `msys2` questions. I will learn eventually :smile:)

**Comment by gillins** [comment]
`/bin/sh: C:Minicondaconda-bldrecipe_1481580663401_b_envpython.exe: command not found`
doesn't look good (see https://ci.appveyor.com/project/conda-forge/gdal-feedstock/build/1.0.273/job/x53k0t31lywciopt#L4421)

Needs quoting or something here. This could get painful...

**Comment by gillins** [comment]
OK it seems that the `GNUmakefile` uses the $PYTHON environment variable if it exists - then obviously there is a quoting problem with `msys2`. I wonder if you can do something like this to workaround it:
```
cd swig\python
set OLDPYTHON=%PYTHON%
set PYTHON=python
make veryclean
make generate
set PYTHON=%OLDPYTHON%
```
If we can't assume that the `python` on the %PATH% is the one to use then we may have to capture the output of `cygpath %PYTHON%` or something....

**Comment by mingwandroid** [comment]
The problem is not quoting, because no amount of quotes will fix it. What happens is that bash interprets the standard Windows dirsep `\` as the escape character. More than that, the msys2 POSIX subsystem has a different filesystem namespace (it has for example `/etc/fstab`)

Because of this, when calling any POSIX program (make.exe from `m2-make` in this case) from bld.bat you need to convert any env. vars that will be interpreted in the POSIX filesystem namespace to that namespace (in the other direction, MSYS2 does this for you automatically though `MSYS2_ENV_CONV_EXCL` can be used to selectively disable that).

You need to use the following to convert things that contain filepaths:
```
for /F "delims=" %%i in ('cygpath.exe -u %PREFIX%') do set "PREFIX=%%i"
for /F "delims=" %%i in ('cygpath.exe -u -p "%LIB%"') do set "LIB=%%i"
```
`-u` means convert to the unix filesystem namespace.
`-p` means convert a pathlist, i.e. a `;` delimited string, e.g. `C:\Some\Dir;C:\Another\Dir`

Because `cygpath.exe` is a POSIX program it understands the MSYS2/Cygwin mounts in `/etc/fstab` and does the right thing with them.

Alternatively, you might be able to use `mingw32-make` from the `m2w64-make` package instead. This is a native port of GNU make which knows nothing about the POSIX filesystem and contains some hacks to make it attempt to understand Windows paths a bit better. It doesn't work fantastically well in all cases, but if you try it and it fails, I can probably advise. FWIW, I never use `mingw32-make`, preferring `m2-make` and converting filepaths instead.

**Comment by ocefpaf** [comment]
Thanks @gillins and @mingwandroid for the info. Now I understand much better what to expected when using `posix`.

However, I believe #129 is on the right track, so I am closing this one in favor of that.

**Comment by gillins** [comment]
Yes thanks @mingwandroid - very informative! Makes a lot of sense.

**Pull request closed without merging**